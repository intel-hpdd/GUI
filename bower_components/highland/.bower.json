{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}