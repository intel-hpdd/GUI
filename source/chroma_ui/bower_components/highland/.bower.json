{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}