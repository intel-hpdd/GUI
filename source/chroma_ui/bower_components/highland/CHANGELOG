Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
