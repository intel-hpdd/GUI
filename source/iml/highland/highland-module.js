// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import highland from 'highland';

export default angular
  .module('highland', [])
  .value('Î»', highland)
  .value('highland', highland).name;
