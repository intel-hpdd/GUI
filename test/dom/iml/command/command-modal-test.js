// @flow

import { render } from "inferno";
import highland from "highland";

describe("command modal", () => {
  let data, commands, CommandModalComponent, cb, div, body, component, jobStream, session$, modalStack$;
  let mockSocketStream, mockMoment, mockGetStore;

  beforeEach(() => {
    data = {
      objects: [
        {
          children: [],
          available_transitions: [
            {
              label: "Cancel",
              state: "cancelled"
            }
          ],
          cancelled: false,
          class_name: "StartTargetJob",
          commands: ["/api/command/230/"],
          created_at: "2019-03-05T17:23:14.773266",
          description: "Start target MGS",
          errored: false,
          id: 523,
          modified_at: "2019-03-05T17:23:14.773240",
          read_locks: [],
          resource_uri: "/api/job/523/",
          state: "tasked",
          step_results: {
            "/api/step/951/": null
          },
          steps: ["/api/step/951/", "/api/step/952/"],
          wait_for: [],
          write_locks: []
        }
      ]
    };
    jobStream = highland();
    mockSocketStream = jest.fn(url => {
      if (url === "/job") return jobStream;
    });
    jest.mock("../../../../source/iml/socket/socket-stream.js", () => mockSocketStream);

    mockMoment = jest.fn(() => ({
      format: jest.fn(() => "Jan 01 2019 00:00:00")
    }));
    jest.mock("moment", () => mockMoment);

    session$ = highland();
    modalStack$ = highland();
    mockGetStore = {
      select: jest.fn(name => {
        if (name === "session") return session$;
        else if (name === "modalStack") return modalStack$;
      }),
      dispatch: jest.fn()
    };
    jest.mock("../../../../source/iml/store/get-store.js", () => mockGetStore);

    CommandModalComponent = require("../../../../source/iml/command/command-modal.js").CommandModalComponent;

    session$.write({
      session: {
        user: "user"
      }
    });

    modalStack$.write(["COMMAND_MODAL_NAME"]);

    commands = [
      {
        cancelled: false,
        complete: true,
        created_at: "created_at",
        errored: false,
        id: 7,
        jobs: ["/api/job/1"],
        logs: "logs",
        message: "message",
        state: "state",
        resource_uri: "resource_uri"
      }
    ];
    cb = jest.fn();

    body = document.querySelector("body");
    div = document.createElement("div");

    if (body != null) body.appendChild(div);

    component = <CommandModalComponent commands={commands} closeCb={cb} />;
    render(component, div);
  });

  it("should render the component", () => {
    jobStream.write(data);
    expect(div).toMatchSnapshot();

    render(null, div);
    if (body) body.removeChild(div);
  });

  describe("on destroy", () => {
    beforeEach(() => {
      cb.mockImplementation(() => {
        render(null, div);
        if (body) body.removeChild(div);
      });
      window.dispatchEvent(new KeyboardEvent("keydown", { key: "Escape" }));
    });

    it("should dispatch with the MODAL_STACK_REMOVE_MODAL type", () => {
      expect(mockGetStore.dispatch).lastCalledWith({
        type: "MODAL_STACK_REMOVE_MODAL",
        payload: "COMMAND_MODAL_NAME"
      });
    });
  });
});
