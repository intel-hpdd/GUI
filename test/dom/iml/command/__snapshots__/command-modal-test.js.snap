// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command modal should render the component 1`] = `
<div>
  <div
    class="modal-open"
  >
    
    <div
      class="modal fade in command-modal"
      role="dialog"
      style="display: block; z-index: 1050;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h3
              class="modal-title"
            >
              Commands
            </h3>
          </div>
          <div
            class="modal-body command-modal-body"
          >
            <div
              aria-multiselectable="true"
              class="panel-group"
              id="command-accordion"
              role="tablist"
            >
              <div
                class="panel panel-default"
              >
                <div
                  class="panel-heading"
                  id="heading0"
                  role="tab"
                >
                  <h4
                    class="panel-title"
                  >
                    <a
                      aria-controls="collapse0"
                      aria-expanded="true"
                      data-parent="#command-accordion"
                      data-toggle="collapse"
                      role="button"
                    >
                      <span>
                        <i
                          class="fa fa-chevron-down"
                        />
                        <i
                          class="fa header-status"
                        />
                        <span>
                          message
                           - 
                          Jan 01 2019 00:00:00
                        </span>
                      </span>
                    </a>
                  </h4>
                </div>
                <div
                  aria-labelledby="heading0"
                  class="panel-collapse collapse in"
                  id="collapse0"
                  role="tabpanel"
                >
                  <div
                    class="panel-body"
                  >
                    <h4>
                      Details:
                    </h4>
                    <table
                      class="table"
                    >
                      <tbody>
                        <tr>
                          <td>
                            Created At
                          </td>
                          <td>
                            Jan 01 2019 00:00:00
                          </td>
                        </tr>
                        <tr>
                          <td>
                            Status
                          </td>
                          <td>
                            State
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <div>
                      <h4>
                        Jobs
                      </h4>
                      <div
                        class="well jobs"
                      >
                        <div>
                          <div
                            class="child-job"
                          >
                            <a>
                              <span
                                class="job-state"
                              >
                                <i
                                  class="fa fa-refresh fa-spin"
                                />
                              </span>
                              <span>
                                Start target MGS
                              </span>
                            </a>
                            
                          </div>
                        </div>
                      </div>
                    </div>
                    <div>
                      <h4>
                        Logs
                      </h4>
                      <pre
                        class="logs"
                      >
                        logs
                      </pre>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-danger"
            >
              Close 
              <i
                class="fa fa-times-circle-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="modal-backdrop fade in command-modal-backdrop"
      style="z-index: 1049;"
    />
  </div>
</div>
`;
